### PR TITLE
Removed old addSettingsUpdateConsumer method

### PR DESCRIPTION
**Pull request by qchimera**
### Description
_Removed old addSettingsUpdateConsumer(Map<Setting<?>, Consumer<?>> settingsUpdateConsumers) method in SDKClusterSettings and actuallized tests to the addSettingsUpdateConsumer(Setting<T> setting, Consumer<T> settingsUpdateConsumer) method._

### Issues Resolved
_#512_ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/test/java/org/opensearch/sdk/TestSDKClusterService.java
Comment (dbwiddis) [inline src/test/java/org/opensearch/sdk/TestSDKClusterService.java:109]: While this works, it ends up duplicating the previous `testAddSettingsUpdateConsumer()` test method.  So this whole test method can be removed.

**Review by dbwiddis — CHANGES_REQUESTED** [review]
LGTM with the noted comment.

Also we need to hold off on merging this until all usages of this method in the Anomaly Detection project [`feature/extension` branch](https://github.com/opensearch-project/anomaly-detection/tree/feature/extensions) have been migrated to the non-map version.   If you'd like to take a shot at editing those now it would help get this PR merged sooner.

File: src/test/java/org/opensearch/sdk/TestSDKClusterService.java
Comment (qchimera) [reply src/test/java/org/opensearch/sdk/TestSDKClusterService.java:109]: Ok, will update this and make a new PR to the [feature/extension](https://github.com/opensearch-project/anomaly-detection/tree/feature/extensions)

**Comment by qchimera** [comment]
haven't found any usage of map-method in 

[anomaly-detection](https://github.com/opensearch-project/anomaly-detection).
Maybe it was removed previously? Or am I missing something?

**Comment by dbwiddis** [comment]
> haven't found any usage of map-method in
> 
> [anomaly-detection](https://github.com/opensearch-project/anomaly-detection). Maybe it was removed previously? Or am I missing something?

You need to look at the `feature/extensions` branch.  There are three cases.  You can remove the try/finally for those while you're at it; looking at the `main` branch for those settings has the syntax that should work.

Screenshot 2023-03-05 at 12 52 23 PM

**Review by dbwiddis — DISMISSED** [review]
LGTM pending:
 - run `./gradlew spotlessApply`
 - fix the 3 usages on AD `feature/extensions` branch (make sure your PR is against that branch)

**Comment by dbwiddis** [comment]
> > * fix the 3 usages on AD
> 
> Should I create an issue for this? Or are there already any?

We are tracking `feature/extension` issues on this SDK repo so the existing issue #512 is fine for this, it's even mentioned there. :)

**Comment by dbwiddis** [comment]
> > * fix the 3 usages on AD
> 
> Should I create an issue for this? Or are there already any?

When you do create a PR there please add `[Feature/extensions]` to your title and make sure it's against that branch (if the diff is too large that'll be obvious!)

**Review by dbwiddis — APPROVED** [review]

**Comment by dbwiddis** [comment]
@qchimera looks like you didn't do DCO signoff on your last commit...

**Comment by dbwiddis** [comment]
> @dbwiddis Well I re-checked and everything is fine for me I guess, logs are: image image Link [here](https://github.com/opensearch-project/opensearch-sdk-java/pull/522/checks?check_run_id=11797190752)

Huh! Thanks for that.  I'll set the test to pass and follow up with these other issues elsewhere.

**Review by saratvemulapalli — DISMISSED** [review]

**Comment by saratvemulapalli** [comment]
@dbwiddis looks like we have taken care of everything. Good to merge?

**Comment by dbwiddis** [comment]
> @dbwiddis looks like we have taken care of everything. Good to merge?

This will break AD until https://github.com/opensearch-project/anomaly-detection/pull/834 is merged.  That PR is blocked by my work on #503.   

SO we have two options:
 - add null checks to https://github.com/opensearch-project/anomaly-detection/pull/834 temporarily and remove them when #503 is done
 - leave this approved and wait for me to address #503 (this week) and merge later.

**Comment by dbwiddis** [comment]
Note: PR #559 will address #503. It may create a small merge conflict with the to-be-deleted test class when it's merged, if so I'll touch up the PR.

**Comment by owaiskazi19** [comment]
@qchimera can you rebase your branch with latest `main`? Thanks

**Comment by dbwiddis** [comment]
I'm going to try to get this merged next week, btw.

**Comment by owaiskazi19** [comment]
@qchimera @dbwiddis do we need a companion PR on AD plugin for the same?

**Review by joshpalis — APPROVED** [review]

**Review by owaiskazi19 — APPROVED** [review]

**Pull request merged**
